addons/im_livechat/static/src/core/web/livechat_channel_model.js

```diff
@@ -20,5 +25,39 @@ export class LivechatChannel extends Record {
     id;
     /** @type {string} */
     name;
+    hasSelfAsMember = false;
+    threads = Record.many("Thread", { inverse: "livechatChannel" });
+
+    async join() {
+        this.hasSelfAsMember = true;
+        if (!this._store.discuss.isActive) {
```

Comment:
I would make it a condition (disabling it only when actually clicking in sidebar)